### ISSUE TITLE
wrong solving status on infeasible mip when giving mip start

### ISSUE THREAD

**Issue by svigerske** [ERGO-Code/HiGHS#902]
When I ask HiGHS 1.2.2 (but also happens on master) to solve an infeasible MIP and pass it an (infeasible) solution, it returns with a Status "Optimal", Solution status "infeasible", and an error `MIP solver claims optimality, but with num/sum/max primal(1/1/1) infeasibilities`:
```
HiGHS 1.2.2 [3f0a74587]
Presolving model
38 rows, 22 cols, 281 nonzeros
33 rows, 17 cols, 129 nonzeros
33 rows, 17 cols, 129 nonzeros

MIP start solution is infeasible, objective value is 177.525

Solving MIP model with:
   33 rows
   17 cols (10 binary, 0 integer, 0 implied int., 7 continuous)
   129 nonzeros

( 0.0s) Starting symmetry detection
( 0.0s) No symmetry present

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work      
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   153.225         inf                  inf        0      0      0         0     0.0s
         0       0         0   0.00%   153.675         inf                  inf        0      0      5        10     0.0s

Solving report
  Status            Optimal
  Primal bound      inf
  Dual bound        inf
  Gap               inf
  Solution status   infeasible
                    177.525 (objective)
                    0 (bound viol.)
                    0 (int. viol.)
                    1 (row viol.)
  Timing            0.54 (total)
                    0.00 (presolve)
                    0.00 (postsolve)
  Nodes             2
  LP iterations     1641 (total)
                    102 (strong br.)
                    369 (separation)
                    1132 (heuristics)
ERROR:   MIP solver claims optimality, but with num/sum/max primal(1/1/1) infeasibilities
```
I don't see how to reproduce this by just giving you input files (the HiGHS exe doesn't allow to pass a file with starting point, or does it?), but hope you'll be able to find the issue anyway. Looks like when it decides on the status, it forgets to check that the solution on record is actually feasible.

**Comment by jajhall** [comment]
Thanks. It should be easy to reproduce this in a unit test

**Comment by jajhall** [comment]
I tried to set up a trivial infeasible MIP, but didn't reproduce the behaviour. If you send me the MIP,  and your starting point dumped into a text file,I can run it

**Comment by svigerske** [comment]
[highs.lp.txt](https://github.com/ERGO-Code/HiGHS/files/9002398/highs.lp.txt)
[highs.mps.txt](https://github.com/ERGO-Code/HiGHS/files/9002399/highs.mps.txt)

The actual solution values probably are not relevant, but anyway, this is what I pass in `sol.col_value`:
```
75
0
275
300
300
0
0
0
50
0
0
1
0
1
1
1
0
0
0
1
0
0
```

**Comment by jajhall** [comment]
I've reproduced this - and corrected my trivial example so that it also exposes the same bug!

**Comment by tasseff** [comment]
Hi, all. Please see the above issue I've posted for SciPy. It appears that, on the HiGHS side, this may have been corrected, but I'm not completely sure. The example problem I posted does appear to return correct solution statuses using the `bin/highs` executable compiled from the master branch of HiGHS.

**Comment by jajhall** [comment]
According to @lgottwald , this issue and #907 were fixed by #913. Happy to create a version with this fix for you to use as a reference
